### ISSUE TITLE
Address overlap check can give false positive for sparse arrays

### ISSUE THREAD

**Issue by andrei-gabriel-blanaru** [SystemRDL/systemrdl-compiler#139]
Attempting to use the empty space between the elements of a sparse array is detected as an address overlap.
E.g.
```
addrmap top {
    regfile example {
        reg some_reg { field {} a; };
        some_reg b[10] @0x100 += 0x10; // These consume 160-12 bytes of space
                                       // Address 0x100 to 0x103, 0x110 to 0x113,....
        some_reg c @0x104; // Incorrectly detected as an overlap
    } exrf;
};
```

**Comment by amykyta3** [comment]
The SystemRDL spec does not do a good job of explaining this, but my understanding is that interleaving arrays is not allowed. According to some discussions I have had with Accelera committee members, SystemRDL tracks closely with the semantics of IP-XACT, including the addressing rules. IP-XACT expressly forbids this type of overlapping (IEEE Std 1685-2014, SCR 7.1)
